### PR TITLE
[SPARK-5336][YARN]spark.executor.cores must not be less than spark.task.cpus

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
https://issues.apache.org/jira/browse/SPARK-5336

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ package org.apache.spark.deploy.yarn
 
 import scala.collection.mutable.ArrayBuffer
 
-import org.apache.spark.SparkConf
+import org.apache.spark.{SparkException, SparkConf}
```
Comment (sryza) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala]: Nit: import ordering should be alphabetical

**Comment by sryza** [comment]
We should warn about this in standalone mode and mesos coarse grained mode as well, right?

**Comment by WangTaoTheTonic** [comment]
@sryza 
In standalone and mesos mode it is a bit different. JIRA is here:https://issues.apache.org/jira/browse/SPARK-5337
Seems like @CodingCat has created a PR: https://github.com/apache/spark/pull/4129, we can discuss the details there.

**Comment by WangTaoTheTonic** [comment]
Actually when cluster has not enough cores, the driver will also hang up. I made a trivial change here and hope it would not make much noise for this pr.

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala [hunk 2]
```diff
@@ -95,6 +95,10 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (executorCores < sparkConf.getInt("spark.task.cpus", 1)) {
+      throw new SparkException("spark.executor.cores must not be less than " +
```
Comment (tgravescs) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala]: only nit here is that I might not have specified it via spark.executor.cores but rather via the spark-submit --executor-cores option.  Perhaps we should just say "Executor cores must".

**Comment by WangTaoTheTonic** [comment]
@tgravescs Updated as you said. Perhaps we should do the same change in `ExecutorAllocationManager`, but I am not familiar with that future, so let's leave it with guys who is in charge of that.

**Comment by tgravescs** [comment]
I agree, the text was that way in ExecutorAllocationManager so lets leave it for now.

This looks good.  Thanks @WangTaoTheTonic

**Pull request closed without merging**